fslock_plan9.go

```diff
@@ -0,0 +1,33 @@
+package fslock
+
+import "strings"
+
+// Opening an exclusive-use file returns an error.
+// The expected error strings are:
+//
+//  - "open/create -- file is locked" (cwfs, kfs)
+//  - "exclusive lock" (fossil)
+//  - "exclusive use file already open" (ramfs)
+//
+// See https://github.com/golang/go/blob/go1.15rc1/src/cmd/go/internal/lockedfile/lockedfile_plan9.go#L16
+var lockedErrStrings = [...]string{
```

Comment:
Agree that the structure you've added here seems like a reasonable way to extend the lock function to plan9, and this follows the definitions in the standard library, so seems fine.

The one semantic thing I'm not clear on that is worth checking if you haven't yet, is that these errors will cover both the "someone else has the lock" error that they will trigger the overall `Lock` method returning, and the `lock is already held by us` case in the main lock switch statement.

I haven't looked to see if any of the callers of `Lock` do different things based on those errors, and my quick read at how plan9 is structuring its errors doesn't leave me overly optimistic that we'd be able to efficiently learn if the existing exclusive fid is in the same process, but it's probably worth checking to make sure there isn't code that's using those semantics.

Reply:
If the lock is held by us, we get [this error](https://github.com/go4org/go4/blob/f5505b9728ddf920bb673137648788c5ac99de1b/lock/lock.go#L55):
```
file "/tmp/filelock/my-test.lock" already locked
```
If the lock is held by another process, we get [this error](https://github.com/go4org/go4/blob/f5505b9728ddf920bb673137648788c5ac99de1b/lock/lock_plan9.go#L37):
```
cannot acquire lock: Lock Create of /tmp/filelock/my-test.lock failed: open /tmp/filelock/my-test.lock: '/tmp/filelock/my-test.lock' file is locked
```
I think I already distinguish between these two errors by checking if the error string contains "Lock Create of" (similar to `isLockCreatePermFail` function). We depends on `go4.org/lock` returning two kinds of errors, but I think it's good enough solution.